sdk/core/azure-core/azure/core/rest/_helpers.py

```diff
@@ -114,8 +114,8 @@ def set_urlencoded_body(data, has_files):
 
 def set_multipart_body(files: FilesType):
     file_items = files.items() if isinstance(files, Mapping) else files
-    formatted_files = {f: _format_data_helper(d) for f, d in file_items if d is not None}
-    return {}, formatted_files
+    formatted_files = [(f, _format_data_helper(d)) for f, d in file_items if d is not None]
+    return {}, dict(formatted_files) if isinstance(files, Mapping) else formatted_files
```

Comment:
And I advise move aiohttp part and test of my PR [[core] Support multi files of same field name for multipart/form-data by msyyc · Pull Request #33853 · Azure/azure-sdk-for-python (github.com)](https://github.com/Azure/azure-sdk-for-python/pull/33853) to yours so that we make sure both async and sync work